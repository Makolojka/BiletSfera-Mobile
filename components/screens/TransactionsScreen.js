import React, { useState, useEffect } from 'react';
import {View, Text, ScrollView, StyleSheet, TouchableOpacity} from 'react-native';
import DataService from "../../services/DataService";
import {COLORS} from "./Colors";
import {formatDate} from "../../utils/DataHelper";

const TransactionsScreen = (props) => {
    const [transactions, setTransactions] = useState([]);

    useEffect(() => {
        async function fetchTransactions() {
            try {
                const transactionsData = await DataService.getAllTransactions();
                const updatedTransactions = await Promise.all(
                    transactionsData.map(async (transaction) => {
                        const ticketDetails = await Promise.all(
                            transaction.tickets.map(async (ticket) => {
                                const ticketDetail = await DataService.getTicketDetailsById(ticket.ticketId);
                                return {
                                    count: ticket.count,
                                    ticketName: ticketDetail.type,
                                    ticketPrice: ticketDetail.price,
                                    eventId: ticket.eventId,
                                    ticketId: ticket.ticketId
                                };
                            })
                        );
                        return { ...transaction, ticketDetails };
                    })
                );
                setTransactions(updatedTransactions);
            } catch (error) {
                console.error('Error fetching transactions:', error);
            }
        }

        fetchTransactions();
    }, []);

    const navigateToTicketDetails = (ticketObj, transactionObj) => {
        // console.log("ticketObj eventId: ",ticketObj.eventId)
        // console.log("transactionObj: ",transactionObj)
        props.navigation.navigate('TicketDetails', { ticket: ticketObj, transaction: transactionObj });
    };

    return (
        <ScrollView contentContainerStyle={styles.container}>
            <Text style={styles.title}>Historia zakupów</Text>
            {transactions?.map((transaction, index) => (
                <View key={index} style={styles.transactionItem}>
                    <Text style={styles.transactionDate}>{formatDate(transaction?.saleDate)}</Text>
                    <Text style={styles.text}>Całkowity koszt zamówienia: {transaction?.totalCost} zł</Text>
                    <Text style={{marginTop: 10, color: COLORS.third, fontWeight: 'bold'}}>Zakupione bilety:</Text>
                    <View>
                        {transaction?.ticketDetails?.map((ticket, idx) => (
                            <View key={idx} style={styles.ticketDetail}>
                                <Text style={styles.text}>{ticket?.count} bilet(ów) {ticket?.ticketName} - {ticket?.ticketPrice} zł</Text>
                                <TouchableOpacity
                                    style={styles.button}
                                    onPress={() => navigateToTicketDetails(ticket, transaction)}
                                >
                                    <Text style={styles.buttonText}>Szczegóły</Text>
                                </TouchableOpacity>
                            </View>
                        ))}
                    </View>
                </View>
            ))}
        </ScrollView>
    );
};

const styles = StyleSheet.create({
    container: {
        flexGrow: 1,
        padding: 20,
    },
    title: {
        fontSize: 20,
        // borderBottomWidth: 1,
        // paddingBottom: 20,
        marginBottom: 20,
        color: COLORS.third,
        // borderBottomColor: COLORS.third,
        textTransform: 'uppercase',
        fontWeight: 'bold',
        textAlign: 'left',
    },
    transactionItem: {
        marginBottom: 20,
        borderWidth: 1,
        borderColor: COLORS.third,
        padding: 10,
        elevation: 1,
    },
    transactionDate: {
        fontWeight: 'bold',
        marginBottom: 5,
        color: COLORS.third
    },
    ticketDetail: {
        marginBottom: 5,
    },
    text: {
        color: 'black'
    },
    button:{
        borderWidth: 1,
        borderColor: COLORS.third,
        backgroundColor: COLORS.second,
        textAlign: "center",
        alignItems: 'center',
        paddingTop: 8,
        paddingBottom: 8,
        margin: 10,
    },
    buttonText:{
        color: COLORS.main,
        textTransform: 'uppercase',
        fontSize: 12,
        fontWeight: "bold",
    }
});

export default TransactionsScreen;