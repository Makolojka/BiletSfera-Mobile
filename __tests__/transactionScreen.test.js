import React from 'react';
import renderer from 'react-test-renderer';
import TransactionsScreen from "../components/screens/TransactionsScreen";
jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
test('renders correctly', () => {
    const tree = renderer.create(<TransactionsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});

