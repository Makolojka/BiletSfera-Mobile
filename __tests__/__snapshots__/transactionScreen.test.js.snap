// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
      "padding": 0,
    }
  }
>
  <View>
    <Text
      style={
        {
          "borderBottomWidth": 1,
          "color": "#2d1a42",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 10,
          "marginLeft": 0,
          "padding": 20,
          "paddingBottom": 10,
          "paddingLeft": 20,
          "textAlign": "left",
          "textTransform": "uppercase",
        }
      }
    >
      Historia zakup√≥w
    </Text>
  </View>
</RCTScrollView>
`;
